src/widgets/settingspages/GeneralPage.hpp

```diff
@@ -93,6 +95,36 @@ class SettingsLayout : public QVBoxLayout
         return button;
     }
 
+    template <typename T>
+    ColorButton *addColorButton(const QString &text, const QColor &color,
+                                pajlada::Settings::Setting<T> &setting)
+    {
+        auto colorButton = new ColorButton(color);
+        auto layout = new QHBoxLayout();
+        auto label = new QLabel(text + ":");
+        layout->addWidget(label);
+        layout->addStretch(1);
+        layout->addWidget(colorButton);
+        this->addLayout(layout);
+        QObject::connect(
+            colorButton, &ColorButton::clicked, [setting, colorButton]() {
+                auto dialog = new ColorPickerDialog(QColor(setting));
+                dialog->setAttribute(Qt::WA_DeleteOnClose);
+                dialog->show();
+                dialog->closed.connect([=] {
+                    QColor selected = dialog->selectedColor();
+
+                    if (selected.isValid())
+                    {
+                        getSettings()->lastMessageColor = selected.name();
```

Comment:
`QColor::name()` does not respect alpha value of the color. Is that intended?

Alternatively, `QColor::name(QColor::HexArgb)` can be used to also save the alpha value.